src/Language/PureScript/TypeChecker/Monad.hs

```diff
@@ -66,7 +66,7 @@ data CheckState = CheckState
   -- ^ The next skolem variable
   , checkNextSkolemScope :: Int
   -- ^ The next skolem scope constant
-  , checkCurrentModule :: Maybe ModuleName
+  , checkCurrentModule :: Maybe (ModuleName, [(SourceAnn, ModuleName, ImportDeclarationType, Maybe ModuleName)])
```

Comment:
This feels like an odd place to add this new data. Why not put a list of `(ModuleName, ImportDeclarationType)` in a new field in `Environment` instead?

Reply:
An `imports :: Map ModuleName (ModuleName, ImportDeclaration)` map in the environment could be useful for other things but it would incur a circular dependency between `Language.PureScript.Environment` and `Language.PureScript.AST.Declarations`.

Reply:
I got confused because `Language.PureScript.Sugar.desugar` accepts a list of modules and I didn’t realize we typecheck only one module at a time so I thought it would be better to keep together data specific to the current module.

It seems that since https://github.com/purescript/purescript/pull/1501/commits/6d1b5ec3af12812810701b766a12f7c9aceabe75 we also only ever desugar singleton lists of modules. Should we update `desugar` from `[Module] -> m [Module]` to `Module -> m Module`? This is straightforward to do and it would simplify a few things.

Reply:
> Should we update `desugar` from `[Module] -> m [Module]` to `Module -> m Module`?

That sounds like a good idea to me; I couldn't dig up a good reason to keep the current signature.

Probably that should be its own PR, though? I'd want to clean up `desugar` itself, `partiallyDesugar` in `L.P.Docs.Convert`, and the few `desugar` steps that aren't simply `map`s or `traverse`s—`desugarImports` has a doc comment that needs to be brought up to date too.

Reply:
Changing desugar so that it only desugars one module at a time also sounds good to me, and I agree that that's probably best done in its own PR.

Reply:
I have a commit ready, I‘ll open another pull request.